src/datastore/src/userstore.js

```diff
@@ -33,6 +35,54 @@ class UserStore extends NetworkStore {
   }
 
   /**
+   * Find all users. A query can be optionally provided to return
+   * a subset of all users or omitted to return all users.
+   *
+   * @param {Query} [query] Query used to filter entities.
+   * @param {Object} [options] Options
+   * @param {Boolean} [options.discover] Discover users.
+   * @return {Observable} Observable.
+   */
+  find(query, options = {}) {
+    // If `options.discover`, use
+    // [User Discovery](http://devcenter.kinvey.com/guides/users#lookup)
+    // instead of querying the user namespace directly.
+    if (options.discover === true) {
```

Comment:
In 1.x the [lookup](https://github.com/Kinvey/js-sdk/blob/1.x/src/core/user.js#L566) was performed using `user.find`. I think it makes more sense to separate it out but this is why I did it this way.

As for what a find does on the user collection, I assume it performs a find like it would on any other collection. Again looking at 1.x you will see that it contains code to perform a regular find. https://github.com/Kinvey/js-sdk/blob/1.x/src/core/user.js#L581.